src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+    }
+
+    // boiler plate code for running simple sga modules
+    private static SGAUnitRuntimeInfoStruct simpleModuleRun(final Path sgaPath,
+                                                            final String moduleName,
+                                                            final String extensionToAppend,
+                                                            final File inputFASTAFile,
+                                                            final File outputDirectory,
+                                                            final int threads,
+                                                            final SGAModule indexer,
+                                                            final ArrayList<String> indexerArgs){
+
+        final SGAModule module = new SGAModule(moduleName);
+        final ArrayList<String> args = new ArrayList<>();
+        args.add("--threads");   args.add(Integer.toString(threads));
+        args.add(inputFASTAFile.getName());
+        final Tuple3<Integer, String, String> moduleInfo = module.run(sgaPath, outputDirectory, args);
+
+        final String outputFileName = extractBaseNameWithoutExtension(inputFASTAFile) + extensionToAppend;
+
+        final SGAUnitRuntimeInfoStruct unitInfo = new SGAUnitRuntimeInfoStruct(moduleName, moduleInfo, outputFileName);
+
+        indexerArgs.set(indexerArgs.size()-1, outputFileName);
+        final Tuple3<Integer, String, String> indexerInfo = indexer.run(sgaPath, outputDirectory, indexerArgs);
+        unitInfo.add("index", indexerInfo);
+
+        return unitInfo;
+    }
+
+    // From https://www.stackoverflow.com/questions/4545937
+    @VisibleForTesting
+    static String extractBaseNameWithoutExtension(final File file){
+        final String[] tokens = file.getName().split("\\.(?=[^\\.]+$)");
+        return tokens[0];
+    }
+
+    /**
+     * A struct to represent the runtime information returned by running a unit of sga modules.
+     * Because some of the modules are designed to be run together in serial form in a unit (e.g. correction followed by indexing)
+     *   all return status and stdout stderr messages in this unit are logged.
+     * But only one file name is important for consumption by downstream units/modules, so only that file name is kept.
+     * The ctor and the add() method are designed asymmetrically, because when a unit is consisted of multiple modules,
+     *   only one generates the file name to be returned, and the other modules are simply to be logged.
+     */
+    @VisibleForTesting
+    static final class SGAUnitRuntimeInfoStruct implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final String fileNameToReturn;
+        public final List<Tuple2<String, Tuple3<Integer, String, String>>> runtimeInfo;
+
+        public SGAUnitRuntimeInfoStruct(final String moduleName, final Tuple3<Integer, String, String> entry, final String fileName){
```

Comment:
To me this would be a lot more readable if you created this by explicitly passing in the elements of the entry (return code, stderr, stdout) instead of a Tuple3. You could create the Tuple3 in the constructor here if that's how you want to store them, but then at least clients don't have to guess at what the elements of the entries should be (as opposed to using named parameters).